frontend/src/components/story/StoryPeriod.vue

```diff
@@ -23,28 +24,50 @@ export default {
       expandedDays: [],
     }
   },
+  computed: {
+    sortedDays() {
+      if (!this.period.days()._meta.loading) {
+        return sortBy(this.period.days().items, (day) => day.dayOffset)
+      }
+      return []
+    },
+  },
   watch: {
     period: {
       immediate: true,
-      handler(value) {
-        this.computeExpandedDays(value)
+      async handler(newPeriod, oldPeriod) {
+        const newPeriodId = newPeriod == undefined ? null : newPeriod.id
+        const oldPeriodId = oldPeriod == undefined ? null : oldPeriod.id
+        if (newPeriodId != oldPeriodId) {
+          await this.updateComponentData(newPeriod)
+        }
```

Comment:
You could use the URIs and optional chaining to simplify this:
```suggestion
         if (newPeriod?._meta.self !== oldPeriod?._meta.self) { 
           await this.updateComponentData(newPeriod) 
         }
```